client/src/Authorize.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import axios from 'axios';
+
+const localStorageKeys = {
+  accessToken: 'spotifyAccessToken',
+  refreshToken: 'spotifyRefreshToken',
+  expireTime: 'tokenExpireTime',
+  timestamp: 'tokenTimestamp'
+};
+
+export default class Authorize extends React.Component {
+
+  componentDidMount() {
+
+    const accessToken = this.props.params.get('access_token');
+    const refreshToken = this.props.params.get('refresh_token');
+    const expiresIn = this.props.params.get('expires_in');
+
+    if (!accessToken || !refreshToken || !expiresIn) {
+      return this.props.onAuthorized(null);
+    }
+    if (window.localStorage.spotifyRefreshToken === 'undefined' || ((Date.now() - Number(window.localStorage.tokenTimestamp)) / 1000) < 1000) {
+      const data = axios.get(`/refresh_token?refresh_token=${window.localStorage.spotifyRefreshToken}`);
+
+      window.localStorage.setItem(localStorageKeys.accessToken, data.access_token);
+      window.localStorage.setItem(localStorageKeys.timestamp, Date.now());
+      window.location.reload();
+    }
```

Comment:
I don't really understand this section. If the refresh token is undefined, why would we send it in a request?

If the server is sending us the token, we should be able to update local storage and then call `this.props.onAuthorized`, no? `window.location.reload()` is almost never something you actually want to do.

Reply:
Oh right, I wrote this one beforehand because I was going to use the condition in order to log the user out since if there is no refresh token we would be out of luck. Forgot to remove it.